lib/drab/utils.ex

```diff
@@ -0,0 +1,161 @@
+
+      defhandler check_answer(socket, sender, value) do
+        decoded_value = Drab.Utils.decode_value(value)
+        
+        question = decoded_value["question"]
+        answer = decoded_value["answer"]
+
+      end
+  """
+  def encode_value(value, options \\ []) do
+    # Options
+    encode = Keyword.get(options, :encode, true)
+    encrypt = Keyword.get(options, :encrypt, false)
+
+    value
+    |> (&((encode || encrypt) && encode_js(&1) || &1)).()
+    |> (&(encrypt && encrypt_value(&1, options) || &1)).()
+    |> (&((encode || encrypt) && Base.encode64(&1, padding: false) || &1)).()
+  end
+
+  @doc """
+  Decode a value encoded with encode_value/2
+
+  ### Options
+
+  * `decode` - `Boolean`, Decode JSON/Base64, default `true`
+  * `decrypt` - `Boolean`, Decrypt the value, default `false`. When `true`, implies `decode` = `true`
+
+  Examples:
+        iex> Drab.Utils.decode_value("eyJtZXNzYWdlIjoiSGVsbG8sIFdvcmxkISJ9")
+        %{"message" => "Hello, World!"}
+  """
+  def decode_value(value, options \\ [])
+  def decode_value(nil,  _options) do "" end
+  def decode_value("",   _options) do "" end
+  def decode_value(value, options) do
+    # options
+    decode = Keyword.get(options, :decode, true)
+    decrypt = Keyword.get(options, :decrypt, false)
+
+    value
+    |> (&((decode || decrypt) && Base.decode64!(&1, padding: false) || &1)).()
+    |> (&(decrypt && decrypt_value(&1, options) || &1)).()
+    |> (&((decode || decrypt) && decode_js(&1) || &1)).()
+  end
+
+  @doc """
+  Extract a specific cookie from cookies string.
+
+  ### Parameters
+  * `cookies` - The string that cotains the cookies
+  * `key` - The name of the cookie to extract
+
+  ### Options
+   See the options for `decode_value`
+
+  """
+  def extract_cookie(cookies, key, options \\ [])
+  def extract_cookie(_cookies, nil, _options) do "" end
+  def extract_cookie(_cookies, "", _options) do "" end
+  def extract_cookie(cookies, key,   options) do
+    cookies
+    |> extract_cookie_string(key)
+    |> extract_cookie_value(key)
+    |> decode_value(options)
+  end
+
+  @doc """
+  Convert raw cookies string in a list of maps, where :key is the cookie name, and :value is the cookie value.
+
+  As at this level it is not possible to know which are the cookies values that have be encoded, their values are the same as those in the original string.
+
+  Examples:
+        iex> Drab.Utils.extract_cookies_maps("_ga=GA1.1.12345.54321; _gid=GA1.1.12345.54321; map1=eyJtZXNzYWdlIjoiSGVsbG8sIFdvcmxkIDEhIn0; _gat_gtag_UA_123ABC=1; cookiebar=CookieAllowed")
+        [
+          %{key: "_ga", value: "GA1.1.12345.54321"},
+          %{key: "_gid", value: "GA1.1.12345.54321"},
+          %{key: "map1", value: "eyJtZXNzYWdlIjoiSGVsbG8sIFdvcmxkIDEhIn0"},
+          %{key: "_gat_gtag_UA_123ABC", value: "1"},
+          %{key: "cookiebar", value: "CookieAllowed"}
+        ]
+  """
+  @spec extract_cookies_maps(String.t()) :: Keyword.t()
+  def extract_cookies_maps(cookies) do
```

Comment:
Again, let's not do this function part of the API. User already has an excellent function to read cookies, why to confuse him with more functions?

Public API should have: cookie, cookies and set_cookie.

Reply:
Yes, less is more! :)